client/components/Calendar.jsx

```diff
@@ -14,26 +14,57 @@ const Calendar = (props) => {
     [0, 0, 0, 0, 0, 0, 0],
     [0, 0, 0, 0, 0, 0, 0],
   ];
-  // get data for first day of current month
-  const first = new Date(props.year, props.month, 1);
+
+  const monthFirstDay = new Date(props.year, props.month, 1);
+  const monthLastDay = new Date(props.year, props.month + 1, 0);
+  const currentMonth = new Date().getMonth();
+
+  const bookingsMonthIndex = props.month < currentMonth ? currentMonth + props.month : props.month - currentMonth;
+  let bookedDates = props.bookedDates[bookingsMonthIndex].slice().sort((a, b) => b - a);
+  //declare counter to fill table with dates
+  let dateCounter = 1;
   
-  //get data for last day of current month
-  const last = new Date(props.year, props.month + 1, 0);
+  const renderDateCell = function (currentCell) {
```

Comment:
Perhaps change 'currentCell' to 'cellValue' to make it clearer that we're not expecting a reference/index for a cell?